application/frontend/controllers/MethodController.php

```diff
@@ -38,6 +40,159 @@ public function actionIndex()
     {
         /** @var User $user */
         $user = \Yii::$app->user->identity;
-        return $user->methods;
+        return $user->getVerifiedMethods();
+    }
+
+    /**
+     * View single method
+     * @param string $uid
+     * @return Method
+     * @throws NotFoundHttpException
+     */
+    public function actionView($uid)
+    {
+        $method = Method::findOne(['uid' => $uid, 'user_id' => \Yii::$app->user->getId()]);
+        if ($method === null) {
+            throw new NotFoundHttpException();
+        }
+
+        return $method;
+    }
+
+    /**
+     * Create new unverified method and send verification
+     * @return Method
+     * @throws BadRequestHttpException
+     * @throws \Exception
+     */
+    public function actionCreate()
+    {
+        $request = \Yii::$app->request;
+
+        $type = $request->post('type');
+        $value = $request->post('value');
+
+        if ($type === null || ! in_array($type, [Method::TYPE_EMAIL, Method::TYPE_PHONE])) {
+            throw new BadRequestHttpException(
+                sprintf('Type is required. Options are: %s or %s', Method::TYPE_EMAIL, Method::TYPE_PHONE)
+            );
+        } elseif ($value === null) {
+            throw new BadRequestHttpException('Value is required.');
+        }
+
+        /*
+         * Check for existing method with this value
+         */
+        $method = Method::findOne(['value' => $value, 'user_id' => \Yii::$app->user->getId()]);
```

Comment:
I dont think that would matter since an email address and phone number cannot be equal, it would not get confused here.